src/testing/native-debugging.md

```diff
@@ -0,0 +1,459 @@
+        if (!await launchUrl(
+          url,
+          mode: LaunchMode.externalApplication,
+        )) {
+          throw Exception('Could not launch $url');
+        }
+      }
+
+      Future<void> _launchInWebView(Uri url) async {
+        if (!await launchUrl(
+          url,
+          mode: LaunchMode.inAppWebView,
+        )) {
+          throw Exception('Could not launch $url');
+        }
+      }
+
+      Widget _launchStatus(BuildContext context, AsyncSnapshot<void> snapshot) {
+        if (snapshot.hasError) {
+          return Text('Error: ${snapshot.error}');
+        } else {
+          return const Text('');
+        }
+      }
+
+      @override
+      Widget build(BuildContext context) {
+        final Uri toLaunch = Uri(
+            scheme: 'https',
+            host: 'docs.flutter.dev',
+            path: 'testing/oem-debuggers');
+        return Scaffold(
+          appBar: AppBar(
+            title: Text(widget.title),
+          ),
+          body: Center(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                Padding(
+                  padding: const EdgeInsets.all(16),
+                  child: Text(toLaunch.toString()),
+                ),
+                FilledButton(
+                  onPressed: () => setState(() {
+                    _launched = _launchInBrowser(toLaunch);
+                  }),
+                  child: const Text('Launch in browser'),
+                ),
+                const Padding(padding: EdgeInsets.all(16)),
+                FilledButton(
+                  onPressed: () => setState(() {
+                    _launched = _launchInWebView(toLaunch);
+                  }),
+                  child: const Text('Launch in app'),
+                ),
+                const Padding(padding: EdgeInsets.all(16.0)),
+                FutureBuilder<void>(future: _launched, builder: _launchStatus),
+              ],
+            ),
+          ),
+        );
+      }
+    }
+    ```
+
+    </details>
+
+1. To add the `url_launcher` dependency, run `flutter pub add`:
+
+    ```terminal
+    $ flutter pub add url_launcher
+    ```
+
+    ```terminal
+    Resolving dependencies... 
+      collection 1.17.1 (1.17.2 available)
+    + flutter_web_plugins 0.0.0 from sdk flutter
+      matcher 0.12.15 (0.12.16 available)
+      material_color_utilities 0.2.0 (0.8.0 available)
+    + plugin_platform_interface 2.1.4
+      source_span 1.9.1 (1.10.0 available)
+      stream_channel 2.1.1 (2.1.2 available)
+      test_api 0.5.1 (0.6.1 available)
+    + url_launcher 6.1.11
+    + url_launcher_android 6.0.36
+    + url_launcher_ios 6.1.4
+    + url_launcher_linux 3.0.5
+    + url_launcher_macos 3.0.5
+    + url_launcher_platform_interface 2.1.3
+    + url_launcher_web 2.0.17
+    + url_launcher_windows 3.0.6
+    Changed 10 dependencies!
+    ```
+
+1. To check what changed with the codebase, run `git status`.
```

Comment:
I'm not sure this step make sense. Unless I missed something, the project they created won't be a git repository yet. Perhaps just the mention about what changes is sufficient?